config/application.rb

```diff
@@ -52,6 +57,15 @@ class Application < Rails::Application
     # Don't log certain requests that spam the log files
     config.middleware.insert_before Rails::Rack::Logger, QuietLogger, paths: ["/healthcheck"]
 
+    # don't fail tests in this case
+    config.active_record.raise_on_assign_to_attr_readonly = false
```

Comment:
This feels slightly dirty but I assume this is something new in Rails 7.1?

Reply:
7.0 fails silently if readonly attributes are written; 7.1 raises an error. Some of our tests do write to readonly attributes. We should probably not do that, though, but presumably we expect the attribute not to change otherwise the tests would have been failing.